src/ucp/wireup/wireup_cm.c

```diff
@@ -1396,7 +1405,7 @@ ucs_status_t ucp_ep_cm_connect_server_lane(ucp_ep_h ep,
     ucs_assert(ucp_ep_get_lane(ep, lane) == NULL);
 
     /* TODO: split CM and wireup lanes */
-    status = ucp_wireup_ep_create(ep, &uct_ep);
+    status = ucp_wireup_ep_create(ep, remote_address, addr_indices, &uct_ep);
```

Comment:
instead of passing both remote_address and addr_indices to ucp_wireup_ep_create, pass only "dst_addr_indices" - after calling ucp_wireup_get_dst_rsc_indices() to calculate them to a local array
this way we would pass minimal information to ucp_wireup_ep_create constructor

Reply:
to calculate `addr_indices` we should do `ucp_wireup_select_lanes` once again

Reply:
done as suggested by @yosefe above